pmlc/rt/runtime.h

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020, Intel Corporation
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <stdexcept>
+#include <utility>
+
+namespace pmlc::rt {
+
+// Device represents a PlaidML device, capable of evaluating a PlaidML program.
+class Device {
+public:
+  // Returns the current Device, local to the current thread.
+  //
+  // If there is no current device, or the current device is of an unexpected
+  // type, an exception will be thrown.
+  //
+  // This is used to smuggle the current device through to runtimes that are
+  // directly invoked by lowered code, when that lowered code doesn't support
+  // the use of a passthrough parameter.
+  template <typename T>
+  static std::shared_ptr<T> current() {
```

Comment:
I think this will go away in the next set of PRs, because it's not really the current device that we need to remember, it's actually the whole initialized state of the vulkan runtime which along with this will have a device to be bound to.

Reply:
I agree -- talking w/Jeremy, it sounds like we're all on board with moving the Vulkan calls out of the lowered code, making it trivial to know what the correct device is.